src/crypto/keybackup.ts

```diff
@@ -35,11 +35,13 @@ export interface IKeyBackupRoomSessions {
 export interface IKeyBackupInfo {
     algorithm: string;
     auth_data: {
-        public_key: string;
-        signatures: ISignatures;
-        private_key_salt: string;
-        private_key_iterations: number;
+        public_key?: string;
+        signatures?: ISignatures;
+        private_key_salt?: string;
+        private_key_iterations?: number;
         private_key_bits?: number;
+        iv?: string;
+        mac?: string;
```

Comment:
Can we not have a discriminated union or XOR here? As presumably there's two different payloads here and not a bunch of fields that can all occur at once.